aea/cli/local_registry_sync.py

```diff
@@ -0,0 +1,140 @@
+
+from aea.cli.fingerprint import determine_package_type_for_directory
+from aea.cli.registry.add import fetch_package
+from aea.cli.registry.utils import get_package_meta
+from aea.cli.utils.context import Context
+from aea.cli.utils.decorators import pass_ctx
+from aea.configurations.data_types import PackageId, PackageType, PublicId
+from aea.configurations.loader import load_component_configuration
+
+
+PACKAGES_DIRS = [i.to_plural() for i in PackageType]
+
+
+@click.command()
+@pass_ctx
+def local_registry_sync(ctx: Context):
+    """Upgrade the local package registry."""
+    do_local_registry_update(ctx.cwd)
+
+
+def do_local_registry_update(base_dir: Union[str, Path]) -> None:
+    """
+    Perform local registry update.
+
+    :param base_dir: root directory of the local registry.
+
+    :return: None
+    """
+    for package_id, package_dir in enlist_packages(base_dir):
+        current_public_id = package_id.public_id
+        latest_public_id = get_package_latest_public_id(package_id)
+        if not (  # pylint: disable=superfluous-parens
+            current_public_id < latest_public_id
+        ):
+            click.echo(f"{package_id} is the latest version.")
+            continue
+        click.echo(f"Updating to {latest_public_id}.")
+        replace_package(str(package_id.package_type), latest_public_id, package_dir)
+
+
+def replace_package(
+    package_type: str, public_id: PublicId, package_dir: Union[Path, str]
+) -> None:
+    """
+    Download, extract and replace exists package.
+
+    :param package_type: str.
+    :param public_id: pacakge bulic id to download
+    :param: package_dir: target package dir
+
+    :return: None
+    """
+    with TemporaryDirectory() as tmp_dir:
+        new_package_dir = os.path.join(tmp_dir, public_id.name)
+        os.mkdir(new_package_dir)
+        fetch_package(
+            package_type, public_id=public_id, cwd=tmp_dir, dest=new_package_dir
+        )
+        shutil.rmtree(package_dir)
+        shutil.move(new_package_dir, package_dir)
+
+
+def get_package_latest_public_id(package_id: PackageId) -> PublicId:
+    """
+    Get package latest package id from the remote repo.
+
+    :param package_id: id of the package to check
+
+    :return: package id of the latest package in remote repo
+    """
+    package_meta = get_package_meta(
+        str(package_id.package_type), package_id.public_id.to_latest()
+    )
+    return PublicId.from_str(package_meta["public_id"])
+
+
+def enlist_packages(
+    base_dir: Union[Path, str]
+) -> Generator[Tuple[PackageId, Union[Path, str]], None, None]:
+    """
+    Generate list of the packages in local repo directory.
+
+    :param base_dir: path or str of the local repo.
+
+    :return: generator with Tuple of package_id and package diurectory.
+    """
+    for author in os.listdir(base_dir):
+        author_dir = os.path.join(base_dir, author)
+        if not os.path.isdir(author_dir):
+            continue  # pragma: nocover
```

Comment:
We should raise an error when `packages` is of invalid format (i.e. invalid directories)

Reply:
(unless the file is an init file, then we skip as you do)